### PR TITLE
Convert to better conventions and fix password validation

### PR DESCRIPTION
**Pull request by a--hoang**
THERE WASN"T VALIDATION? WTF

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -56,16 +56,17 @@ module.exports = function (passport) {
   }, function (req, email, password, done) {
     req.flash('submittedEmail', email)
     User.findOne({ 'local.email': email }, function (err, user) {
-      if (user && !err) return done(null, user)
```
Comment (a--hoang) [inline config/passport.js:59]: holy ballsacks

File: config/passport.js [hunk 2]
```diff
@@ -56,16 +56,17 @@ module.exports = function (passport) {
...
+      if (err) { return done(err) }
+      if (user && user.validPassword(password)) { return done(null, user) }
```
Comment (a--hoang) [inline config/passport.js:60]: @NicoHinderling you didn't implement password validation last time lmao

**Comment by NicoHinderling** [comment]
holy fuck

**Comment by NicoHinderling** [comment]
https://github.com/Bostonhacks/headstart/commit/82e8f98376eadaab968f991bde1fe577beb7c7c2#diff-c6601fa9f2ae68b5ce23cfd416544f27L53
 ............. i can't even.........

**Pull request merged**